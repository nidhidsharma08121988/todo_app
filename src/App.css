@import url('https://fonts.googleapis.com/css2?family=Roboto+Mono:wght@300&display=swap');

* {
  box-sizing: border-box;
}
:root {
  --primary-color: steelblue;
  --primary-color-darker: #396a93;
  --primary-color-darkest: #24435c;
  --secondary-color: black;
  --secondary-color-light: #282c35;
  --secondary-color-lighter: #363636;
  --secondary-color-lightest: #454b4b;
  --light-color: lightgrey;
  --danger-color: red;
  --stuck-task: #780909;
  --success-color: green;
  --completed-task: green;
  --doing: orange;
  --primary-bg: #fdfdfd;
}

body {
  margin: 0;
  display: flex;
  flex-direction: column;
  font-family: 'Roboto Mono', monospace;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 100vw;
  height: 100vh;
}
/* container */
.container {
  display: flex;
  flex-direction: column;
  max-width: 100%;
  align-items: center;
}
/* sub-container */
.sub-container {
  display: flex;
  flex-direction: column;
  max-width: 100%;
  justify-content: top;
  align-items: center;
}

/* Navbar */
.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-shadow: 1px 2px 3px var(--primary-color-darker);
  background-color: var(--primary-color-darkest);
  color: white;
  width: 100vw;
}
.navbar ul {
  list-style: none;
  padding-left: 0;
  display: inline-flex;
}
.left-navbar {
  display: inline-flex;
}

/* tasks based borders */
.task-completed {
  border-left: 4px solid var(--completed-task);
}
.task-ongoing {
  border-left: 4px solid var(--doing);
}
.task-stuck {
  border-left: 4px solid var(--stuck-task);
}
.sub-task-completed {
  color: var(--completed-task);
}
.sub-task-ongoing {
  color: var(--doing);
}
.sub-task-stuck {
  color: var(--stuck-task);
}
/* ul which lists all tasks */
ul {
  list-style: none;
  padding-left: none;
  margin-left: 0px;
}
.sub-tasks {
  list-style: none;
  padding-left: 0px;
  color: var(--light-color);
}
.sub-task {
  border-radius: 20px;
}
/* links */
a,
Link {
  color: white;
  text-decoration: none;
}
a:visited,
Link:visited {
  text-decoration: none;
}

a:hover,
Link:hover {
  font-weight: 900;
}

/* button styles */
.float-right {
  float: right;
}
.tooltip {
  display: none;
  opacity: 0.8;
  margin-top: 0.1rem;
  background-color: var(--primary-color-darkest);
  color: white;
}

.btn-tooltip-div:hover .tooltip {
  display: flex;
  position: absolute;
  z-index: 2;
}

.btn {
  display: inline-block;
  font-size: 1.5rem;
  border: none;
  cursor: pointer;
  padding: 0.4rem 1.3rem;
  margin-right: 0.5rem;
  transition: opacity 0.2s ease-in;
  outline: none;
  text-align: center;
}

.add-btn {
  display: inline-flex;
  background-color: var(--primary-color-darkest);
  color: white;
  border-radius: 3px;
  box-shadow: 3px 3px 8px var(--light-color);
}

.btn-sm {
  font-size: 0.8rem;
  padding: 0.3rem 1rem;
  margin-right: 0.2rem;
}

.btn-link {
  background: none;
  padding: 0;
  margin: 0;
}
.btn-smooth {
  border-radius: 25px;
}
.btn-primary {
  width: 100%;
}
.btn-block {
  display: block;
  width: 100%;
}
.badge {
  font-size: 0.8rem;
  padding: 0.2rem 0.7rem;
  text-align: center;
  margin: 0.3rem;
  background-color: var(--light-color);
  color: #333;
  border-radius: 5px;
}
.alert {
  padding: 0.7rem;
  margin: 1rem 0;
  opacity: 0.9;
  background-color: var(--light-color);
  color: #333;
}

.btn-primary,
.bg-primary,
.badge-primary,
.alert-primary {
  background-color: var(--primary-color-darkest);
  color: #fff;
}

.btn-light,
.bg-light,
.badge-light,
.alert-light {
  background-color: var(--light-color);
  color: #333;
}

.btn-dark,
.bg-dark,
.badge-dark,
.alert-dark {
  background-color: var(--secondary-color-light);
  color: #fff;
}

.btn-danger,
.bg-danger,
.badge-danger,
.alert-danger {
  background-color: var(--danger-color);
  color: #fff;
}

.btn-success,
.bg-success,
.badge-success,
.alert-success {
  background: var(--success-color);
  color: #fff;
}

.btn-white,
.bg-white,
.badge-white,
.alert-white {
  background: #fff;
  color: #333;
  border: #ccc solid 1px;
}

.btn:hover {
  opacity: 0.8;
}

.bg-light,
.badge-light {
  border: #ccc solid 1px;
}

/* form */
.form-container {
  display: flex;
  flex-direction: column;
  margin-top: 3px;
  border: 0.5px dotted var(--light-color);
  box-shadow: 1px 1px 2px var(--light-color);
  padding: 3px;
  width: 100%;
  max-width: 500px;
  align-items: left;
  justify-content: center;
}
.form-controls {
  margin-top: 1px;
  margin-bottom: 1px;
  display: flex;
  flex-direction: column;
  width: 100%;
}
.form-control {
  display: flex;
  flex-direction: row;
  margin: 2px;
  padding: 2px;
  width: 100%;
}
textarea {
  height: 200px;
  width: 100%;
  max-width: 100%;
  max-height: 200px;
  scroll-behavior: auto;
  resize: none;
}
.input {
  width: 100%;
}
/* border */
.bl {
  border-left: 0.5px dotted var(--secondary-color-lightest);
}
.bb {
  border-bottom: 0.5px dotted var(--secondary-color-lightest);
}

/* card */
.card {
  margin: 2px;
  border: 0.3px dotted darkgrey;
  border-radius: 20px;
  box-shadow: 1px 2px 3px darkgrey;
  max-width: 100%;
  width: 700px;
  display: flex;
  flex-direction: column;
  background-color: var(--primary-bg);
}
/* heading */
.heading {
  color: var(--primary-color-darkest);
}
/* sub heading */
.sub-heading {
  color: var(--primary-color-darker);
}
/* content */
.content {
  color: var(--secondary-color-light);
}
/* texts */
.text-xs {
  font-size: 0.3rem;
}
.text-sm {
  font-size: 0.5rem;
}
.text-reg {
  font-size: 1rem;
}
.text-l {
  font-size: 1.2rem;
}
.text-xl {
  font-size: 1.8rem;
}
/* margin */
.m-1 {
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}
.m-2 {
  margin-left: 1rem;
  margin-right: 1rem;
}
.m-3 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}
.my-1 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}
.my-2 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}
.my-3 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}
/* padding */
.p-1 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}
.p-2 {
  padding-left: 1rem;
  padding-right: 1rem;
}
.p-3 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}
.py-1 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}
.py-2 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}
.py-3 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

/* icons */
i {
  color: white;
}
.i-l {
  color: var(--light-color);
}

/* based on display */
.inline-flex {
  display: inline-flex;
}
.flex-row {
  display: flex;
}
.flex-col {
  display: flex;
  flex-direction: column;
}
.space-between {
  justify-content: space-between;
}
.center {
  justify-content: center;
  align-items: center;
}

/* mobile styles settings */
@media (max-width: 700px) {
  * {
    box-sizing: border-box;
  }
  body {
    font-size: 10px;
    scroll-behavior: auto;
    width: 100vw;
    max-width: 600px;
    height: 100vh;
  }
  .container {
    width: 100%;
    max-width: 500px;
  }
  .sub-container {
    width: 95vw;
  }
  .card {
    width: 100%;
  }
  .task-dates {
    display: flex;
    flex-direction: column;
  }
  .dates {
    border-left: none;
    padding-top: 0.5rem;
    border-top: 0.5px dotted var(--secondary-color-lightest);
  }
  .form-container {
    max-width: 400px;
  }
}
