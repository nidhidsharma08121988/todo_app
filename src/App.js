import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import './App.css';
import Navbar from './components/layout/Navbar';
import Alert from './components/layout/Alert';
import Home from './components/pages/Home';
import About from './components/pages/About';
import ListState from './context/list/ListState';
import AlertState from './context/alert/AlertState';
import AddTodoForm from './components/todos/AddTodoForm';
import EditTodoForm from './components/todos/EditTodoForm';

function App() {
  return (
    <ListState>
      <AlertState>
        <Router>
          <div className='App'>
            <Navbar />
            <Alert />
            <div className='container py-container'>
              <Switch>
                <Route exact path='/' component={Home} />
                <Route exact path='/about' component={About} />
                <Route exact path='/addTodo' component={AddTodoForm} />
                <Route
                  exact
                  path='/editTodo/:todoItemId'
                  component={EditTodoForm}
                />
              </Switch>
            </div>
          </div>
        </Router>
      </AlertState>
    </ListState>
  );
}

export default App;
