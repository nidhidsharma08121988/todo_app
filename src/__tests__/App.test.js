// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import React from 'react';
import App from '../App';
import { render } from '@testing-library/react';

// test what is visible on screen
describe('App must display properly', () => {
  test('Must match snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test('Must have title', () => {
    const { queryByText } = render(<App />);
    expect(queryByText('Todo')).toBeTruthy();
  });

  test('Must display menu with link home and About us', () => {
    const { queryByText } = render(<App />);
    expect(queryByText('Home')).toBeTruthy();
    expect(queryByText('About us')).toBeTruthy();
  });

  test('Must have Things to do', () => {
    const { queryByText } = render(<App />);
    expect(queryByText('Things to do')).toBeTruthy();
  });

  test('Must have add button on screen', () => {
    const { getByTestId } = render(<App />);
    expect(getByTestId('add-btn')).toBeTruthy();
  });
});

// test the user events produce expected results
describe('Check event interaction', () => {
  test('Clicking add-btn must open Add form', () => {
      
  });
});
