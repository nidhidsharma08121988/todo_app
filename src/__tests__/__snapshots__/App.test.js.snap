// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App must display properly Must match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="navbar p-1"
    >
      <div
        class="left-navbar text-xl"
      >
        <div
          class="p-1"
        >
          <i
            class="fa fa-book"
          />
        </div>
        <div
          class="p-1"
        >
          Todo
        </div>
      </div>
      <div
        class="right-navbar menu-container m-1"
      >
        <i
          class="fa fa-bars menu-btn text-l"
        />
        <ul
          class="menu m-1 text-reg"
        >
          <li
            class="menu-list p-1"
          >
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="menu-list p-1"
          >
            <a
              href="/about"
            >
              About us
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div />
    <div
      class="container py-container"
    >
      <div>
        <div
          class="btn-div"
        >
          <a
            href="/addTodo"
          >
            <button
              class="add-btn btn my-2 p-1"
              data-testid="add-btn"
            >
              + 
              <span
                class="add-btn-hover"
              >
                Add Task
              </span>
            </button>
          </a>
        </div>
        <div
          class="sub-container"
        >
          <div
            class="flex-row space-between"
          >
            <h2
              class="heading text-xl"
            >
              Things to do
            </h2>
          </div>
          <div>
            <ul>
              <div
                class="sub-cards"
              >
                <li>
                  <div
                    class="card my-1 p-2"
                  >
                    <div
                      class="flex-row space-between bb"
                    >
                      <div
                        class="flex-row"
                      >
                        <h3
                          class="sub-heading my-1 py-1 m-1"
                        >
                          Complete this app
                        </h3>
                      </div>
                      <div
                        class="ctr-btn"
                      >
                        <a
                          href="/editTodo/1"
                        >
                          <button
                            class="btn btn-sm btn-dark my-1 py-1 light-smooth"
                          >
                            <i
                              class="fa fa-edit"
                            />
                          </button>
                        </a>
                        <button
                          class="btn btn-sm btn-danger my-1 py-1 light-smooth"
                        >
                          <i
                            class="fa fa-trash"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="task-dates flex-row space-between"
                    >
                      <div
                        class="content"
                      >
                        <ul
                          class="sub-tasks"
                        >
                          <li>
                            <div
                              class="sub-task my-1 p-1"
                            >
                              Implement display list feature
                            </div>
                          </li>
                          <li>
                            <div
                              class="sub-task my-1 p-1"
                            >
                              Implement add to list feature
                            </div>
                          </li>
                          <li>
                            <div
                              class="sub-task my-1 p-1"
                            >
                              Implement delete list item feature
                            </div>
                          </li>
                          <li>
                            <div
                              class="sub-task my-1 p-1"
                            >
                              Implement edit list item feature
                            </div>
                          </li>
                        </ul>
                      </div>
                      <div
                        class="dates flex-col bl m-1 p-1 my-1"
                      >
                        Added on
                        <div
                          class="flex-row my-1"
                        >
                          <i
                            class="fa fa-calendar m-1 i-l"
                          />
                          <div
                            class="inline-flex m-1"
                          >
                            2021-05-18
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="flex-row"
                    >
                      <span
                        class="lbl-disp badge-light m-1 my-1 p-1"
                      >
                        completed
                      </span>
                      <span
                        class="lbl-disp badge-light m-1 my-1 p-1"
                      >
                        fast
                      </span>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    class="card my-1 p-2"
                  >
                    <div
                      class="flex-row space-between bb"
                    >
                      <div
                        class="flex-row"
                      >
                        <h3
                          class="sub-heading my-1 py-1 m-1"
                        >
                          Complete honeypot test
                        </h3>
                      </div>
                      <div
                        class="ctr-btn"
                      >
                        <a
                          href="/editTodo/2"
                        >
                          <button
                            class="btn btn-sm btn-dark my-1 py-1 light-smooth"
                          >
                            <i
                              class="fa fa-edit"
                            />
                          </button>
                        </a>
                        <button
                          class="btn btn-sm btn-danger my-1 py-1 light-smooth"
                        >
                          <i
                            class="fa fa-trash"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="task-dates flex-row space-between"
                    >
                      <div
                        class="content"
                      >
                        <ul
                          class="sub-tasks"
                        >
                          <li>
                            <div
                              class="sub-task my-1 p-1"
                            >
                              Practice algorithms
                            </div>
                          </li>
                          <li>
                            <div
                              class="sub-task my-1 p-1"
                            >
                              Take practice test
                            </div>
                          </li>
                          <li>
                            <div
                              class="sub-task my-1 p-1"
                            >
                              Take test
                            </div>
                          </li>
                        </ul>
                      </div>
                      <div
                        class="dates flex-col bl m-1 p-1 my-1"
                      >
                        Added on
                        <div
                          class="flex-row my-1"
                        >
                          <i
                            class="fa fa-calendar m-1 i-l"
                          />
                          <div
                            class="inline-flex m-1"
                          >
                            2021-04-27
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="flex-row"
                    >
                      <span
                        class="lbl-disp badge-light m-1 my-1 p-1"
                      >
                        completed
                      </span>
                      <span
                        class="lbl-disp badge-light m-1 my-1 p-1"
                      >
                        successfully
                      </span>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    class="card my-1 p-2"
                  >
                    <div
                      class="flex-row space-between bb"
                    >
                      <div
                        class="flex-row"
                      >
                        <h3
                          class="sub-heading my-1 py-1 m-1"
                        >
                          Complete logic test
                        </h3>
                      </div>
                      <div
                        class="ctr-btn"
                      >
                        <a
                          href="/editTodo/3"
                        >
                          <button
                            class="btn btn-sm btn-dark my-1 py-1 light-smooth"
                          >
                            <i
                              class="fa fa-edit"
                            />
                          </button>
                        </a>
                        <button
                          class="btn btn-sm btn-danger my-1 py-1 light-smooth"
                        >
                          <i
                            class="fa fa-trash"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="task-dates flex-row space-between"
                    >
                      <div
                        class="content"
                      >
                        <ul
                          class="sub-tasks"
                        >
                          <li>
                            <div
                              class="sub-task my-1 p-1"
                            >
                              Practice algorithms
                            </div>
                          </li>
                          <li>
                            <div
                              class="sub-task my-1 p-1"
                            >
                              Take practice test
                            </div>
                          </li>
                          <li>
                            <div
                              class="sub-task my-1 p-1"
                            >
                              Take test
                            </div>
                          </li>
                        </ul>
                      </div>
                      <div
                        class="dates flex-col bl m-1 p-1 my-1"
                      >
                        Added on
                        <div
                          class="flex-row my-1"
                        >
                          <i
                            class="fa fa-calendar m-1 i-l"
                          />
                          <div
                            class="inline-flex m-1"
                          >
                            2021-06-1
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="flex-row"
                    >
                      <span
                        class="lbl-disp badge-light m-1 my-1 p-1"
                      >
                        completed
                      </span>
                      <span
                        class="lbl-disp badge-light m-1 my-1 p-1"
                      >
                        successful
                      </span>
                      <span
                        class="lbl-disp badge-light m-1 my-1 p-1"
                      >
                        waiting for result
                      </span>
                    </div>
                  </div>
                </li>
              </div>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
